bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,288 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract superclass of all supported Broadlink devices.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BroadlinkBaseThingHandler extends BaseThingHandler implements DeviceRediscoveryListener {
+
+    public final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String INITIAL_DEVICE_ID = "00000000";
+
+    protected BroadlinkDeviceConfiguration thingConfig;
+
+    private @Nullable RetryableSocket socket;
+
+    // Can be injected for test purposes
+    private @Nullable NetworkTrafficObserver networkTrafficObserver;
+
+    private int count;
+    private @Nullable ScheduledFuture<?> refreshHandle;
+    private boolean authenticated = false;
+    // These get handed to us by the device after successful authentication:
+    private byte[] deviceId;
+    private byte[] deviceKey;
+
+    public BroadlinkBaseThingHandler(Thing thing, Logger logger) {
+        super(thing);
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+        count = (new Random()).nextInt(65535);
+
+        logger.debug("constructed: resetting deviceKey to '{}', length {}",
+                BroadlinkBindingConstants.BROADLINK_AUTH_KEY, BroadlinkBindingConstants.BROADLINK_AUTH_KEY.length());
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+
+        this.socket = new RetryableSocket(thingConfig, logger);
+    }
+
+    // For test purposes
+    void setSocket(RetryableSocket socket) {
+        this.socket = socket;
+    }
+
+    void setNetworkTrafficObserver(NetworkTrafficObserver networkTrafficObserver) {
+        this.networkTrafficObserver = networkTrafficObserver;
+    }
+
+    private boolean hasAuthenticated() {
+        return authenticated;
+    }
+
+    public void initialize() {
+        logger.debug("initializing polling");
+
+        updateItemStatus();
+
+        if (thingConfig.getPollingInterval() != 0) {
+            refreshHandle = scheduler.scheduleWithFixedDelay(new Runnable() {
+
+                public void run() {
+                    updateItemStatus();
+                }
+            }, 1L, thingConfig.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    public void thingUpdated(Thing thing) {
+        forceOffline(ThingStatusDetail.CONFIGURATION_PENDING, "Thing has been updated, will reconnect soon");
+        // Refetch the config NOW before we come back up again...
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+    }
```

Comment:
All this can be removed if you follow the comments above moving the thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class); to the initilise() method. This works too but is more code :)